### DISCUSSION TITLE
WhereCascade的使用方法

### DISCUSSION THREAD

**Discussion by hellosnowy** [dotnetcore/FreeSql#1388] [General]
Iselect的WhereCascade方法有没有具体的使用实例？
我测试只能单表，
多表例如
`fsql.Select<t1,t2,t3,t4....>().LeftJoin(w=>w.t1.id==w.t2.mid)`
，这种查询后面就不能加WhereCascade了？

**Comment by 2881099** [comment]
```c#
var query = fsql.Select<T1>().WhereCascade(a => ...)
    .From<T2, T3, T4>((_, b, c, d) => _)
    .LeftJoin(w => w.t1.id == w. t2.mid)
...
```

**Reply by hellosnowy** [reply]
首先，感谢回复。
为什么我在官网的例子
`fsql.Select<t1>()
  .LeftJoin<t2>(...)
  .WhereCascade(x => x.IsDeleted == false)
  .ToList();`
为什么我多表查询的时候，查询后面只能where、WhereIf、WhereDynamicFilter
我只引用了 PackageReference Include="FreeSql" Version="3.2.682"

**Reply by 2881099** [reply]
多表没有实现 WhereCascade，最初为导航属性的多表设计的。使用频率不高，因此没有继续扩展。

**Reply by hellosnowy** [reply]
好的。感觉这个功能在全库都是逻辑删除，多表联查的时候很有用
